node/nodeLoadAccounts.go

```diff
@@ -34,119 +19,56 @@ func (n *Node) loadUserAccountHandlerByAddress(address string, options api.Accou
 }
 
 func (n *Node) loadUserAccountHandlerByPubKey(pubKey []byte, options api.AccountQueryOptions) (state.UserAccountHandler, api.BlockInfo, error) {
-	blockInfo, err := n.getBlockInfoGivenOptions(options)
-	if err != nil {
-		return nil, api.BlockInfo{}, err
-	}
+	repository := n.stateComponents.AccountsRepository()
 
-	account, err := n.loadUserAccountOnRootHash(pubKey, blockInfo.rootHash)
-	if err != nil {
-		return nil, api.BlockInfo{}, err
-	}
+	var account vmcommon.AccountHandler
+	var accountBlockInfo state.AccountBlockInfo
+	var err error
 
-	userAccount, ok := n.castAccountToUserAccount(account)
-	if !ok {
-		return nil, api.BlockInfo{}, ErrCannotCastAccountHandlerToUserAccountHandler
+	if options.OnFinalBlock {
```

Comment:
this is not the node's job to decide which method to call.

Reply:
Somehow, I felt that the selection criteria belongs to a layer closer to the presentation / API (e.g. thinking of API controllers) - as opposed to the selection implementation, which belongs to e.g. our accounts repository.